skyportal/handlers/api/source.py

```diff
@@ -976,17 +981,69 @@ class Validator(Schema):
                 )
             other = ca.Point(ra=ra, dec=dec)
             obj_query = obj_query.filter(Obj.within(other, radius))
+        if start_date or end_date:
+            if start_date:
+                mjd_start = Time(isoparse(start_date)).mjd
+                start_date = arrow.get(start_date.strip()).datetime
+            else:
+                mjd_start = 0
+
+            if end_date:
+                mjd_end = Time(isoparse(end_date)).mjd
+                end_date = arrow.get(end_date.strip()).datetime
+            else:
+                mjd_end = Time.now().mjd
+                end_date = Time.now().datetime
+
+            if mjd_end - mjd_start < 0:
+                return self.error('endDate must be after startDate')
+
+            # raise if time between start->end is too long
+            # if mjd_end - mjd_start > 31:  # more than one month
+            #     return self.error(f'endDate {end_date} is more than one month from startDate {start_date}. ')
+
+            if query_mode == 'old':  # old method, for comparison
+                obj_query = obj_query.filter(
+                    Obj.last_detected_at(self.current_user) >= start_date
+                )
+                obj_query = obj_query.filter(
+                    Obj.last_detected_at(self.current_user) <= end_date
+                )
+                # new method, to become the only one after validation
+            elif query_mode == 'new':
+                # first get only the objects that have any detections within the time range
+                in_range_subquery = (
+                    Photometry.query_records_accessible_by(self.current_user)
+                    .filter(Photometry.mjd >= mjd_start, Photometry.mjd <= mjd_end)
+                    .filter(Photometry.snr > PHOT_DETECTION_THRESHOLD)
+                    .subquery()
+                )
+
+                obj_query = obj_query.join(
+                    in_range_subquery, Obj.id == in_range_subquery.c.obj_id
+                )
+                # then must also disqualify objects that have detections after the time range
+                after_range_subquery = (
+                    Photometry.query_records_accessible_by(self.current_user)
+                    .filter(Obj.id == Photometry.obj_id)
+                    .filter(Photometry.mjd >= mjd_end)
+                    .filter(Photometry.snr > PHOT_DETECTION_THRESHOLD)
+                    .scalar_subquery()
+                )
+
+                obj_query = obj_query.outerjoin(
+                    after_range_subquery, Obj.id == after_range_subquery.c.obj_id
+                ).filter(after_range_subquery.c.id.is_(None))
```

Comment:
@mcoughlin @stefanv Maybe one of you knows how to do a proper `scalar_subquery` off of an existing query? 
I'm trying to get all objects that don't have even one photometry points after end_date.